### ISSUE TITLE
Cucumber exclude query tests fails

### ISSUE THREAD

**Issue by fijemax** [Project-OSRM/osrm-backend#4770]
I realized two differents cucumber test and both of them fails.
Both test represent exclude scenario : 

In the first test case : 
```
            """
            a               g
            .               .
            b....d-$-$-e....f
            .               .
            c               h
            """
```
The segment 'de' is is tagged as 'toll'
Result : All route tests shouldn't be success because 'de' is the only way to achieve it, but all results indicate opposite.

The second test case differs from the first only by the 'hi' segment (normal segment without toll tag).
```
            """
            a               g
            .               .
            b....d-$-$-e....f
            .               .
            c               h..i
            """
```
This shouldn't change anything to the first result.
Result : All route tests shouldn't be success because 'de' is the only way to achieve it.
This time all tests (excepted 'ga' route) success.

This both test case show anything is wrong with exclude param query.

Here my test : 
```
@routing @testbot @exclude
Feature: Testbot - Exclude flags
    Scenario: 1
        Given the profile "testbot"

        Given the node map
            """
            a               g
            .               .
            b....d-$-$-e....f
            .               .
            c               h
            """

        And the ways
            | nodes | highway  | toll | #               |
            | ab    | primary  |      | always drivable |
            | cb    | primary  |      | always drivable |
            | bd    | primary  |      | always drivable |
            | de    | motorway | yes  | not drivable for exclude=toll |
            | ef    | primary  |      | always drivable |
            | fg    | primary  |      | always drivable |
            | fh    | primary  |      | always drivable |

        Given the query options
            | exclude | toll |

        When I route I should get
            | from | to | route         |
            | a    | h  |               |
            | a    | g  |               |
            | g    | a  |               |
            | d    | e  |               |

    Scenario: 2
        Given the profile "testbot"

        Given the node map
            """
            a               g
            .               .
            b....d-$-$-e....f
            .               .
            c               h..i
            """

        And the ways
            | nodes | highway  | toll | #               |
            | ab    | primary  |      | always drivable |
            | cb    | primary  |      | always drivable |
            | bd    | primary  |      | always drivable |
            | de    | motorway | yes  | not drivable for exclude=toll |
            | ef    | primary  |      | always drivable |
            | fg    | primary  |      | always drivable |
            | fh    | primary  |      | always drivable |
            | hi    | primary  |      | always drivable |

        Given the query options
            | exclude | toll |

        When I route I should get
            | from | to | route         |
            | a    | h  |               |
            | a    | g  |               |
            | g    | a  |               |
            | d    | e  |               |
            | d    | i  |               |
```
and here the result test : 
```
Node Version v6.12.3
@routing @testbot @exclude
Feature: Testbot - Exclude flags

  @routing @testbot @exclude
  Scenario: 1
  ✔ Given the profile "testbot"
  ✔ Given the node map
      """
      a               g
      .               .
      b....d-$-$-e....f
      .               .
      c               h
      """
  ✔ And the ways
      | nodes | highway  | toll | #                             |
      | ab    | primary  |      | always drivable               |
      | cb    | primary  |      | always drivable               |
      | bd    | primary  |      | always drivable               |
      | de    | motorway | yes  | not drivable for exclude=toll |
      | ef    | primary  |      | always drivable               |
      | fg    | primary  |      | always drivable               |
      | fh    | primary  |      | always drivable               |
  ✔ Given the query options
      | exclude | toll |
  ✖ When I route I should get
      | from | to | route |
      | a    | h  |       |
      | a    | g  |       |
      | g    | a  |       |
      | d    | e  |       |

  @routing @testbot @exclude
  Scenario: 2
  ✔ Given the profile "testbot"
  ✔ Given the node map
      """
      a               g
      .               .
      b....d-$-$-e....f
      .               .
      c               h..i
      """
  ✔ And the ways
      | nodes | highway  | toll | #                             |
      | ab    | primary  |      | always drivable               |
      | cb    | primary  |      | always drivable               |
      | bd    | primary  |      | always drivable               |
      | de    | motorway | yes  | not drivable for exclude=toll |
      | ef    | primary  |      | always drivable               |
      | fg    | primary  |      | always drivable               |
      | fh    | primary  |      | always drivable               |
      | hi    | primary  |      | always drivable               |
  ✔ Given the query options
      | exclude | toll |
  ✖ When I route I should get
      | from | to | route |
      | a    | h  |       |
      | a    | g  |       |
      | g    | a  |       |
      | d    | e  |       |
      | d    | i  |       |

Failures:

1) Scenario: 1 - features/testbot/exclude4.feature:3
   Step: When I route I should get - features/testbot/exclude4.feature:28
   Step Definition: features/step_definitions/routing.js:4
   Message:
     Tables were not identical:
     | from | to |     route |
     | a    | h  | (-)       |
     | a    | h  | (+) ab,bd,de,ef,fh,fh |
     | a    | g  | (-)       |
     | a    | g  | (+) ab,bd,de,ef,fg,fg |
     | g    | a  | (-)       |
     | g    | a  | (+) fg,ef,de,bd,ab,ab |
     | d    | e  | (-)       |
     | d    | e  | (+) de,de |

2) Scenario: 2 - features/testbot/exclude4.feature:35
   Step: When I route I should get - features/testbot/exclude4.feature:61
   Step Definition: features/step_definitions/routing.js:4
   Message:
     Tables were not identical:
     | from | to |     route |
     | a    | h  |           |
     | a    | g  |           |
     | g    | a  | (-)       |
     | g    | a  | (+) fg,ef,de,bd,ab,ab |
     | d    | e  |           |
     | d    | i  |           |

2 scenarios (2 failed)
10 steps (2 failed, 8 passed)
0m00.105s
```

**Comment by TheMarex** [comment]
This is indeed weird and seems to work with MLD but not with CH.

**Comment by TheMarex** [comment]
I think I understand the problem now, it happens when there is only a single excluded node in the path. This causes an edge case for the CH code, since edges where one node is not excluded are not filtered.

The fix is a little bit ugly, but I think I know what to do.

**Comment by TheMarex** [comment]
This was fixed with #4783.

**Comment by fijemax** [comment]
Thanks for the fix
